### PR TITLE
PAYARA-3085 OpenAPI Doesn't Support @...Param Annotated Fields

### PR DESCRIPTION
**Pull request by MeroRai**

### CODE REVIEW

File: appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/visitor/OpenApiWalker.java [hunk 1]
```diff
@@ -182,17 +184,33 @@ public void accept(ApiVisitor visitor) {
 
             processAnnotation(clazz, annotationClass, annotationFunction, altClass, altFunction,
                     new OpenApiContext(api, getResourcePath(clazz)));
-
+            List<Field> fields = new ArrayList<>();
             for (Field field : clazz.getDeclaredFields()) {
-                processAnnotation(field, annotationClass, annotationFunction, altClass, altFunction,
```
Comment (MattGill98) [inline appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/visitor/OpenApiWalker.java]: I believe we discussed removing this? This class would be best kept generic, moving the implementation specific logic into the handleParam methods.

File: appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/processor/ApplicationProcessor.java [hunk 2]
```diff
@@ -356,66 +356,86 @@ public void visitConsumes(Consumes consumes, AnnotatedElement element, ApiContex
...
+        }
+        
+        if (context.getWorkingOperation() != null) {
+            context.getWorkingOperation().addParameter(newParameter);
+        } else {
+            Field field = Field.class.cast(element);
+            for (Method method : field.getDeclaringClass().getDeclaredMethods()) {
+                OpenAPI api = context.getApi();
+                ApiContext apiContext = new OpenApiContext(api, null, ModelUtils.getOperation(method, api, generateResourceMapping(classes)));
+                apiContext.getWorkingOperation().addParameter(newParameter);
```
Comment (MattGill98) [inline appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/processor/ApplicationProcessor.java]: I'm confused as to how this works, isn't this context just thrown away once it's used?

**Comment by MeroRai** [comment]
jenkins test please

**Comment by MeroRai** [comment]
jenkins test please

File: appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/processor/ApplicationProcessor.java [hunk 3]
```diff
@@ -416,14 +420,32 @@ public void visitCookieParam(CookieParam param, AnnotatedElement element, ApiCon
...
     
     private void addParameter(AnnotatedElement element, ApiContext context,
             org.eclipse.microprofile.openapi.models.parameters.Parameter newParameter) {
         if (element instanceof java.lang.reflect.Parameter) {
-            newParameter.setSchema(new SchemaImpl().type(ModelUtils
-                    .getSchemaType(java.lang.reflect.Parameter.class.cast(element).getType())));
+            SchemaImpl schema = new SchemaImpl();
+            java.lang.reflect.Parameter parameter = java.lang.reflect.Parameter.class.cast(element);
+            schema.setType(ModelUtils.getSchemaType(parameter.getType()));
+//            newParameter.setSchema(new SchemaImpl().type(ModelUtils
```
Comment (smillidge) [inline appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/processor/ApplicationProcessor.java]: Can you remove the commented out code?

**Pull request merged**